nifi-asymmetrik-standard-bundle/nifi-asymmetrik-standard-reporting-tasks/src/main/java/com/asymmetrik/nifi/reporting/CloudwatchNiFiClusterMetricsReporter.java

```diff
@@ -0,0 +1,267 @@
+package com.asymmetrik.nifi.reporting;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+import com.asymmetrik.nifi.models.SystemMetricsSnapshot;
+import com.google.common.collect.ImmutableList;
+
+import org.apache.nifi.annotation.behavior.DynamicProperty;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.expression.AttributeExpression;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.ReportingContext;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchClientBuilder;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.MetricDatum;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+
+@Tags({"disk", "system", "monitoring", "metrics", "reporting", "aws", "cloudwatch"})
+@DynamicProperty(name = "tag name", value = "tag value", description = "dynamic properties will be converted to dimensions and will be applied to all metrics")
+@CapabilityDescription("Calculates the amount of storage space available for Content and Flowfile Repositories, " +
+        "calculates the total count and size of the queue, and emits these metrics to CloudWatch.")
+
+public class CloudwatchNiFiClusterMetricsReporter extends AbstractNiFiClusterMetricsReporter {
+
+    // This is commented out because including the AWSCredentialsProviderService class in this git repo redefines the service in NiFi.
```

Comment:
Rick had a good idea: that the commented code should be removed, and that a note about the AWSCredentialsProviderService not being supported (using a DefaultAWSCredentialsProviderChain instead) should be placed in additionalDetails.html. Look in this repo for an example of how to do that. The information then is visible in the NiFi UI.